sdk/eidolon_ai_sdk/apu/longterm_memory_unit.py

```diff
@@ -0,0 +1,146 @@
+        return self.mem0.add(message, user_id=user_id, metadata=metadata)
+
+    def storeMessages(self, call_context: CallContext, messages: List[LLMMessage]):
+        user_id = AgentOS.current_user().id
+        agent_name = AgentOS.current_agent_name()
+        metadata = {
+            "process_id": call_context.process_id,
+            "agent_name": agent_name
+        }
+        # not sure if I should manipulate result somehow - what's the expected return value?
+        return self.mem0.add(messages, user_id=user_id, metadata=metadata)
+
+    def searchMemories(self, call_context: CallContext, query: str | LLMMessage | List[LLMMessage]):
+        user_id = AgentOS.current_user().id
+        results = self.mem0.search(query) if self._multi_user_scope() else self.mem0.search(query, user_id=user_id)
+        if (self.unit_scope == LongTermMemoryUnitScope.SYSTEM):
+            return results
+
+        def filter_func(res):
+            try:
+                valid = True
+                if not self._multi_process_scope() and res['metadata']['process_id'] != call_context.process_id:
+                    valid = False
+                if not self._multi_agent_scope() and res['metadata']['agent_name'] != AgentOS.current_agent_name():
+                    valid = False
+                return valid
+            except Exception:
+                # assume caused by missing process_id or agent_id fields
+                if self._multi_process_scope() and self._multi_agent_scope():
+                    return True
+                return False
+
+        filtered_results = list(filter(filter_func, results))
+        return filtered_results
+
+    def getAllMemories(self, call_context: CallContext):
+        if self._unit_scope == LongTermMemoryUnitScope.SYSTEM:
+            return self.mem0.get_all()
+        elif self.unit_scope == LongTermMemoryUnitScope.USER:
+            return self.mem0.get_all(user_id=AgentOS.current_user().id)
+
+        user_id = AgentOS.current_user().id
+        memories = self.mem0.get_all() if self._multi_user_scope() else self.mem0.get_all(user_id=user_id)
+
+        def filter_func(mem):
+            try:
+                valid = True
+                if not self._multi_process_scope() and mem['metadata']['process_id'] != call_context.process_id:
+                    valid = False
+                if not self._multi_agent_scope() and mem['metadata']['agent_name'] != AgentOS.current_agent_name():
+                    valid = False
+                return valid
+            except Exception:
+                if self._multi_agent_scope() and self._multi_process_scope():
+                    return False
+                return True
+
+        return list(filter(filter_func, memories))
+
+    def getMemory(self, memory_id: str):
+        return self.mem0.get(memory_id)
+
+    def getMemoryHistory(self, memory_id: str):
+        return self.mem0.history(memory_id)
+
+    def deleteMemory(self, memory_id: str):
```

Comment:
I don't see us using anything other than searchMemories and messages, do we need the other methods? Since this is the contract for long term memory, keeping it concise now will make expanding it easier.